charger/ocpp/instance.go

```diff
@@ -31,6 +32,7 @@ func Instance() *CS {
 
 		server := ws.NewServer()
 		server.SetTimeoutConfig(timeoutConfig)
+		server.SetCheckOriginHandler(func(r *http.Request) bool { return true })
```

Comment:
Imho this would have security implications? Can we rather validate the URI, maybe stripping the invalid schema? This also needs a comment _why_ we're doing this.

Reply:
The connection is established via http, so you only get to the socket by accepting an http connection - which is not source checked for in evcc but by a potential firewall. That socket than switches from http to a websocket protocol, and here the library checks whether the http messages Origin: field in GET ws://a.b.c.d:888/CHARGER matches the already established connection. The actual tcp socket connection remains the same in all this, so afterwards when a JSON message is sent it goes back the same tcp stream way like the http replies before, so I dont see a security threat. I will add a comment. 
occp-go has: "By default, if the Origin header is present in the request, and the Origin host is not equal to the host request header, the websocket handshake fails." in ./ws/websocket.go. I basically looked into the underlying websocket library (websocket/server.go) and took the same function, that library basically returns true if empy, true if the host is the same and false otherwise, but does not actually support talking back to somewhere else (or to ip vs hostname mismatches) - which most simple firewalls would not support anyway.